torch/lib/THC/generic/THCTensorMathBlas.cu

```diff
@@ -658,8 +658,20 @@ THC_API void THCTensor_(btrifact)(THCState *state, THCTensor *ra_, THCudaIntTens
 
   long num_batches = ra__->size[0];
 
-  THCudaIntTensor_resize2d(state, rpivots_, num_batches, n);
-  int *pivots_gpu = THCudaIntTensor_data(state, rpivots_);
+  if (!pivot) {
+    THCudaIntTensor *t = THCudaIntTensor_new(state);
+    THCudaIntTensor_range(state, t, 1, n, 1);
+    THCudaIntTensor_unsqueeze1d(state, t, t, 0);
+    THCudaIntTensor** ptrs = (THCudaIntTensor**) malloc(sizeof(THCudaIntTensor*)*num_batches);
```

Comment:
We should probably not use `malloc` and `free`, but `THAlloc` and `THFree` instead (or their cuda counterparts).